app/models/content_item.rb

```diff
@@ -45,7 +45,11 @@ class ContentItem < ActiveRecord::Base
   validates :details, well_formed_content_types: { must_include: "text/html" }
 
   def requires_base_path?
-    EMPTY_BASE_PATH_FORMATS.exclude?(document_type)
+    EMPTY_BASE_PATH_FORMATS.exclude?(document_type || format)
+  end
+
+  def pathless?
+    !self.requires_base_path? && !Location.exists?(content_item: self)
```

Comment:
One alternative that I see is that requires_base_path? and pathless? could be moved to Location (and take a content_item as an argument).

I really don't mind either way, both are internally consistent.

WebContentItem does not seem the right place, as I don't see any reason why that should be treated differently to ContentItem when it comes to a separation of concerns (then again, I don't really agree with this "theme" off limited visibility of the content item, I'm hoping this will sort itself out when the database structure improves as a whole).